src/audio/host.c

```diff
@@ -379,6 +383,11 @@ static struct comp_dev *host_new(struct sof_ipc_comp *comp)
 	dma_sg_init(&hd->local.elem_array);
 
 	hd->chan = DMA_CHAN_INVALID;
+
+	hd->buffer_cb.cb = &host_buffer_cb;
```

Comment:
@tlauda The way it's setup now, we don't need any dynamic allocations, as callback objects are members of components. I'm not sure if it's a good idea to allocate callbacks separately, as it would cause memory waste with each allocation overhead.
I agree, that it would be easier to register, and probably would look more pleasing, but I have my concerns about memory. If you think that should be changed anyway, then I'll adjust my PR.